### PR TITLE
Alberta Spouse and common-law partner amount credit

### LINKED ISSUE #283: Alberta Marital Status Tax Credit
Reference: https://kings-printer.alberta.ca/1266.cfm?page=a30.cfm&leg_type=Acts&isbncln=9780779841790

### PR DESCRIPTION
**Pull request by 7qiwx**
Fixes #283

### CODE REVIEW

File: policyengine_canada/variables/household/person/count_dependants.py
Comment (MaxGhenis) [inline policyengine_canada/variables/household/person/count_dependants.py:1]: add a formula from `is_dependant`

File: policyengine_canada/parameters/gov/provinces/ab/benefits/marital_status/base.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+description: Alberta marital status base credit
+values:
+  2023-01-01: 19_369
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta marital status base amount
+  reference:
+  - title: Government of Canada - Alberta marital status base credit
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/ab/benefits/marital_status/base.yaml:9]:
```suggestion
  - title: Government of Canada - Alberta marital status base credit (page 18)
```

File: policyengine_canada/variables/gov/provinces/ab/benefits/marital_status/ab_marital_status_credit.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+class ab_marital_status_credit(Variable):
+    value_type = float
+    entity = Household
+    label = "Alberta marital status credit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.AB
+
+    def formula(household, period, parameters):
+        spouse_income = add(household, period, ["separated_spouse_income"])
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/ab/benefits/marital_status/ab_marital_status_credit.py:12]: I think it's the opposite: you can take a credit for your spouse's income only if not separated

File: policyengine_canada/variables/gov/provinces/ab/benefits/marital_status/separated_spouse_income.py
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/ab/benefits/marital_status/separated_spouse_income.py:1]: Don't think we need this - instead map the spouse's income

File: policyengine_canada/variables/household/demographic/filing_status.py
Comment (MaxGhenis) [inline policyengine_canada/variables/household/demographic/filing_status.py:1]: We don't need this: Canada doesn't have joint filing, each earner files separately

File: policyengine_canada/variables/gov/provinces/ab/benefits/marital_status/ab_marital_status_credit.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        p = parameters(period).gov.provinces.ab.benefits.marital_status
+        return max_(0, (p.base - spouse_income))
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/ab/benefits/marital_status/ab_marital_status_credit.py:14]: This is also multiplied by the "specified percentage for the year" along with other credits right?

File: policyengine_canada/parameters/gov/provinces/ab/benefits/marital_status/base.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+description: Alberta marital status base credit
+values:
+  2023-01-01: 19_369
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/ab/benefits/marital_status/base.yaml:3]: Can you find a more recent document to cite as well, as to confirm that it's not inflation-indexed?

File: policyengine_canada/variables/gov/provinces/ab/benefits/marital_status/ab_marital_status_credit.py (hunk 3 above)
Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/ab/benefits/marital_status/ab_marital_status_credit.py:14]: This just specifies the number of months that the person resided in Alberta

**Pull request merged**